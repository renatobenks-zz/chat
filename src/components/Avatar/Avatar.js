// @flow

import styled from 'styled-components';

const Avatar = styled.img.attrs({
  src: ({ avatar }) => avatar,
})`
  width: ${props => props.size || props.theme.avatar.size};
  height: auto;
  border-radius: 50%;
  border: 2px solid ${props => props.border || props.theme.palette.white};
  margin: 0;
`;

export default Avatar;
