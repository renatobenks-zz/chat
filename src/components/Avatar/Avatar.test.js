import React from 'react';

import { render } from '../../../test/helpers';

import Avatar from './Avatar';

const image = 'image.png';

const renderer = component => render(component, { withTheme: true });

it('avatar with children', () => {
  expect(
    renderer(
      <Avatar avatar={image}>
        <span>anything</span>
      </Avatar>,
    ),
  ).toMatchSnapshot();
});

it('avatar with no children', () => {
  expect(renderer(<Avatar avatar={image} />)).toMatchSnapshot();
});

it('avatar with no image', () => {
  expect(renderer(<Avatar />)).toMatchSnapshot();
});

it('avatar with custom size', () => {
  expect(renderer(<Avatar avatar={image} size="40px" />)).toMatchSnapshot();
});

it('avatar with custom border', () => {
  expect(renderer(<Avatar avatar={image} border="black" />)).toMatchSnapshot();
});
