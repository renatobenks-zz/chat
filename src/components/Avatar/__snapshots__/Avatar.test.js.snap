// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar with children 1`] = `
<img
  className="sc-bdVaJa bPnyOK"
  src="image.png"
>
  <span>
    anything
  </span>
</img>
`;

exports[`avatar with custom border 1`] = `
<img
  className="sc-bdVaJa lazlvv"
  src="image.png"
/>
`;

exports[`avatar with custom size 1`] = `
<img
  className="sc-bdVaJa dhlLGt"
  size="40px"
  src="image.png"
/>
`;

exports[`avatar with no children 1`] = `
<img
  className="sc-bdVaJa bPnyOK"
  src="image.png"
/>
`;

exports[`avatar with no image 1`] = `
<img
  className="sc-bdVaJa bPnyOK"
/>
`;
